comp/core/tagger/impl-remote/remote.go

```diff
@@ -250,6 +256,68 @@ func (t *remoteTagger) LegacyTag(entity string, cardinality types.TagCardinality
 	return t.Tag(entityID, cardinality)
 }
 
+// GenerateContainerIDFromExternalData returns a container ID for the given external data.
+func (t *remoteTagger) GenerateContainerIDFromExternalData(externalData origindetection.ExternalData) (string, error) {
+	expBackoff := backoff.NewExponentialBackOff()
+	expBackoff.InitialInterval = 500 * time.Millisecond
+	expBackoff.MaxInterval = 1 * time.Second
+	expBackoff.MaxElapsedTime = 15 * time.Second
+
+	var containerID string
+
+	err := backoff.Retry(func() error {
+		select {
+		case <-t.telemetryTicker.C:
+			t.store.collectTelemetry()
+		case <-t.ctx.Done():
+			return &backoff.PermanentError{Err: errTaggerFailedGenerateContainerIDFromExternalData}
+		default:
+		}
+
+		t.telemetryStore.ExternalDataRequests.Inc()
+
+		// Fetch the auth token
+		token, err := t.options.TokenFetcher()
```

Comment:
Should we do this on every function call or just do it once?

Reply:
I've added the token to the `remoteTagger` struct